src/test/java/util/DummyDataGenerator.java

```diff
@@ -33,6 +36,30 @@
     private static DeleteFlatRequest DUMMY_DELETE_FLAT_REQUEST;
     private static DeleteMateRequest DUMMY_DELETE_MATE_REQUEST;
 
+    public static User generateDummyUserSingleton() {
```

Comment:
We don't generate a new instance every time the method is called and the clients of this class don't have to they are using singleton objects.
In my opinion method names as getUser() is easier to understand and users of this method should not bother knowing they refer to singletons.

Also I would name the static variables normally as user instead of DUMMY_USER